cmd/token/cmd.go

```diff
@@ -0,0 +1,168 @@
+package token
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/openshift/rosa/pkg/config"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	header    bool
+	payload   bool
+	signature bool
+	refresh   bool
+	generate  bool
+}
+
+var Cmd = &cobra.Command{
+	Use:   "token",
+	Short: "Generates a token",
+	Long:  "Uses the stored credentials to generate a token.",
+	Args:  cobra.NoArgs,
+	Run:   run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.payload,
+		"payload",
+		false,
+		"Print the JSON payload.",
+	)
+	flags.BoolVar(
+		&args.header,
+		"header",
+		false,
+		"Print the JSON header.",
+	)
+	flags.BoolVar(
+		&args.signature,
+		"signature",
+		false,
+		"Print the signature.",
+	)
+	flags.BoolVar(
+		&args.refresh,
+		"refresh",
+		false,
+		"Print the refresh token instead of the access token.",
+	)
+	flags.BoolVar(
+		&args.generate,
+		"generate",
+		false,
+		"Generate a new token.",
+	)
+	output.AddFlag(Cmd)
+}
+
+func run(cmd *cobra.Command, argv []string) {
```

Comment:
@tirthct Same suggestion here about defining a service that performs the logic and simply calling into it from the `cobra.Command`.

Reply:
Added code coverage without mocking or using the service

Reply:
@tirthct Thank you! 

Could you please change this to be the same format as the `get` / `set` commands where we bubble the error upwards and not have lots of `os.Exit(1)` statements e.g:

```
r := rosa.NewRuntime()

err := CreateToken(args)
if err != nil {
	r.Reporter.Errorf(err.Error())
	os.Exit(1)
}
```